### PR TITLE
fix: update second_sign method to use schnorr

### PR DESCRIPTION
**Pull request by galperins4**
## Summary

Update the second sign function to use Schnorr

The current code is broken. First sign is Schnorr and second sign was ECSDA which led transactions having issues being verified since they were not correct

Updated a single function in base.py 

## Checklist

- [X] Documentation _(if necessary)_
- [X] Tests _(if necessary)_
- [X] Ready to be merged

### CODE REVIEW

**Comment by ItsANameToo** [comment]
Can you add a test that includes the second signature so this gets caught in any future versions? Just for transfer will suffice for now.

File: crypto/transactions/builder/base.py [hunk 1]
```diff
@@ -66,6 +67,9 @@ def multi_sign(self, passphrase, index):
 
     def schnorr_verify(self):
         return self.transaction.verify_schnorr()
+    
+    def schnorr_verify_second(self):
```
Comment (ItsANameToo) [inline crypto/transactions/builder/base.py]: this needs to get a `secondPublicKey` parameter that you pass along to the verify function

File: crypto/transactions/transaction.py [hunk 2]
```diff
@@ -162,6 +162,16 @@ def verify_schnorr(self):
 
         return is_valid
 
+    def verify_schnorr_secondsig(self):
```
Comment (ItsANameToo) [inline crypto/transactions/transaction.py]: requires a `secondPublicKey` parameter

File: crypto/transactions/transaction.py [hunk 3]
```diff
@@ -162,6 +162,16 @@ def verify_schnorr(self):
...
+        """Verify the transaction. Method will raise an exception if invalid, if it's valid it will
+        returns True
+        """
+        transaction = sha256(self.to_bytes()).digest()
```
Comment (ItsANameToo) [inline crypto/transactions/transaction.py]: remove this

File: crypto/transactions/transaction.py [hunk 4]
```diff
@@ -162,6 +162,16 @@ def verify_schnorr(self):
...
+        is_valid = schnorr.b410_schnorr_verify(transaction, self.senderPublicKey, self.signSignature)
```
Comment (ItsANameToo) [inline crypto/transactions/transaction.py]:
```suggestion
        is_valid = schnorr.b410_schnorr_verify(self.to_bytes(False, True), secondPublicKey, self.signSignature)
```

**Comment by ItsANameToo** [comment]
After that you can add a tests in `tests/transactions/builder/test_transfer.py` that is similar to the existing transfer test, but will now have a second passphrase added and verified (for verifying you can use something like `transaction.schnorr_verify_second(PublicKey.from_passphrase('this is a top secret passphrase too')`, just make sure it matches the passphrase you used to second sign with)

**Comment by galperins4** [comment]
@ItsANameToo I made all the changes suggested and added the extra test. Feel free to re-review.

File: crypto/transactions/builder/base.py [hunk 5]
```diff
@@ -66,6 +67,10 @@ def multi_sign(self, passphrase, index):
 
     def schnorr_verify(self):
         return self.transaction.verify_schnorr()
+    
+    def schnorr_verify_second(self, passphrase):
+        secondPublicKey = PublicKey.from_passphrase(passphrase)
```
Comment (ItsANameToo) [inline crypto/transactions/builder/base.py]: This needs to take a `secondPublicKey` only as you don't know the `passphrase` of someone else. You can fetch the second public key from an address when it has one, and use that to verify the second signature. Hence why you will have to do the `PublicKey.from_passphrase(passphrase)` step in the test itself to mimic this.

File: tests/transactions/builder/test_transfer.py [hunk 6]
```diff
@@ -27,6 +27,29 @@ def test_transfer_transaction():
...
+    transaction_dict = transaction.to_dict()
+
+    assert transaction_dict['nonce'] == 1
+    assert transaction_dict['signature']
+    assert transaction_dict['signSignature']
+    assert transaction_dict['type'] is TRANSACTION_TRANSFER
+    assert transaction_dict['typeGroup'] == 1
+    assert transaction_dict['typeGroup'] == TRANSACTION_TYPE_GROUP.CORE.value
+
+    transaction.schnorr_verify_second('second top secret passphrase') # if no exception is raised, it means the transaction is valid
```
Comment (ItsANameToo) [inline tests/transactions/builder/test_transfer.py]: please add a `transaction.schnorr_verify` line above it to ensure it validates both signatures

**Comment by galperins4** [comment]
All updated. I think this is ready now.

**Pull request merged**